govcd/media.go

```diff
@@ -354,3 +414,224 @@ func (mediaItem *MediaItem) Refresh() error {
 
 	return err
 }
+
+// Refresh refreshes the media information by href
+func (media *Media) Refresh() error {
+
+	if media.Media == nil {
+		return fmt.Errorf("cannot refresh, Object is empty")
+	}
+
+	url := media.Media.HREF
+
+	// Empty struct before a new unmarshal, otherwise we end up with duplicate
+	// elements in slices.
+	media.Media = &types.Media{}
+
+	_, err := media.client.ExecuteRequest(url, http.MethodGet,
+		"", "error retrieving media: %s", nil, media.Media)
+
+	return err
+}
+
+// GetMediaByHref finds a Media by HREF
+// On success, returns a pointer to the Media structure and a nil error
+// On failure, returns a nil pointer and an error
+func (cat *Catalog) GetMediaByHref(mediaHref string) (*Media, error) {
+
+	media := NewMedia(cat.client)
+
+	_, err := cat.client.ExecuteRequest(mediaHref, http.MethodGet,
+		"", "error retrieving media: %#v", nil, media.Media)
+	if err != nil && strings.Contains(err.Error(), "MajorErrorCode:403") {
+		return nil, ErrorEntityNotFound
+	}
+	if err != nil {
+		return nil, err
+	}
+	return media, nil
+}
+
+// GetMediaByName finds a Media by Name
+// On success, returns a pointer to the Media structure and a nil error
+// On failure, returns a nil pointer and an error
+func (cat *Catalog) GetMediaByName(mediaName string, refresh bool) (*Media, error) {
+	if refresh {
+		err := cat.Refresh()
+		if err != nil {
+			return nil, err
+		}
+	}
+	for _, catalogItems := range cat.Catalog.CatalogItems {
+		for _, catalogItem := range catalogItems.CatalogItem {
+			if catalogItem.Name == mediaName && catalogItem.Type == "application/vnd.vmware.vcloud.catalogItem+xml" {
+				catalogItemElement, err := cat.GetCatalogItemByHref(catalogItem.HREF)
+				if err != nil {
+					return nil, err
+				}
+				return cat.GetMediaByHref(catalogItemElement.CatalogItem.Entity.HREF)
+			}
+		}
+	}
+	return nil, ErrorEntityNotFound
+}
+
+// GetMediaById finds a Media by ID
+// On success, returns a pointer to the Media structure and a nil error
+// On failure, returns a nil pointer and an error
+func (cat *Catalog) GetMediaById(mediaId string, refresh bool) (*Media, error) {
```

Comment:
If the implementation doesn't require looping through the internal items, the refresh argument is not needed.

Reply:
good catch - removed